packages/fuels-signers/src/provider.rs

```diff
@@ -183,7 +193,67 @@ impl Provider {
             metadata: None,
         }
     }
-    // TODO: add unit tests for the balance API. This is tracked in #321.
+
+    /// Craft a transaction used to transfer funds to a contract.
+    pub fn build_contract_transfer_tx(
+        &self,
+        to: ContractId,
+        amount: u64,
+        asset_id: AssetId,
+        inputs: &[Input],
+        outputs: &[Output],
```

Comment:
use `Vec`

Reply:
Slices are more flexible, why require a vec? We don't need ownership here.

Reply:
I get what you mean, but since we use `Vec<something>` pretty much everywhere I figured it would make more sense. But you're right, iirc it would mean that we don't have to `clone` to use that method? 
In that case, maybe we should be using slices more.

Reply:
Ideally, according to the [rust API guidelines](https://rust-lang.github.io/api-guidelines/flexibility.html?highlight=intoiter#functions-minimize-assumptions-about-parameters-by-using-generics-c-generic), we should strive for the least constricting interface -- i.e. the most flexible.

> The fewer assumptions a function makes about its inputs, the more widely usable it becomes.

> Prefer
> 
> ```rust
> fn foo<I: IntoIterator<Item = i64>>(iter: I) { /* ... */ }
> ```
> over any of
> 
> 
> ```rust
> fn foo(c: &[i64]) { /* ... */ }
> fn foo(c: &Vec<i64>) { /* ... */ }
> fn foo(c: &SomeOtherCollection<i64>) { /* ... */ }
> ```
> if the function only needs to iterate over the data.
> 
> More generally, consider using generics to pinpoint the assumptions a function needs to make about its arguments.

It further goes on to list the advantages and disadvantages. Using slices here is the pragmatic solution to using the `IntoIterator` trait bound due to the verbosity of the latter. 

But slices have no inherent increase in verbosity while offering a more flexible interface accepting anything that can be converted into a slice of the given type -- aka Vecs, arrays, etc.